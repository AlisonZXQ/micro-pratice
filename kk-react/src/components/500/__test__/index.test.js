import React from 'react';
import { shallow } from 'enzyme';
import mountTest from '@tests/shared/mountTest';
import Error from '../index';

describe('500', () => {
  mountTest(Error);

  it('should snapshot correctly', () => {
    const shallowWrapper = shallow(<Error />);
    expect(shallowWrapper).toMatchSnapshot();
  });

  it('should have div', () => {
    const shallowWrapper = shallow(<Error />);
    expect(shallowWrapper.find('div').exists()).toBe(true);
  });

  it('should have 4 div tags', () => {
    const shallowWrapper = shallow(<Error />);
    expect(shallowWrapper.find('div').length).toBe(4)
  });
})

