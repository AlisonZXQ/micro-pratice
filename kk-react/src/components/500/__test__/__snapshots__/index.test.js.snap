// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`500 should snapshot correctly 1`] = `
<div
  className="root"
>
  <div
    className="m-error-404 f-cb"
  >
    <div
      className="m-pic f-fl"
    >
      <MyIcon
        className="icon"
        type="icon-500"
      />
    </div>
    <div
      className="m-tip f-fl"
    >
      <h1>
        500
      </h1>
      <h2>
        抱歉，服务器出错了
      </h2>
    </div>
  </div>
</div>
`;
